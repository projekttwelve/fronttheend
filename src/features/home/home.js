import React from 'react';

/*The home component, */
export function Home(){
   return(
     <div>
       <p>"Welcome to komodo, to search for your dream job please login or create an account
         if you have not yet done so." </p>
   </div>);

}
