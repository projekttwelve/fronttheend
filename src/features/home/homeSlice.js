import { createAsyncThunk, createSlice} from '@reduxjs/toolkit';
import axios from 'axios';
//import { client } from '../../api/api'
const initialState ={
  posts: [],
  status: null,
  error: null,
  hasAcc: false
};

export const fetchPosts = createAsyncThunk('home/fetchPosts', async(dispatch, getstate)=>{
  const res = await axios("https://jsonplaceholder.typicode.com/users");
  console.log('res', res)
  return res.data;
})
/*
 * accpets:
 * (1) A string that will be used as the prefix for the generated action types
   (2) A "payload creator" callback function that should return a Promise
   containing some data,  or a rejected Promise with an error
   response -->{data: []},
 */
export const homeSlice= createSlice({
  name: 'home',
  initialState,
  reducers:{
    toggleHasAcc: (state)=>{
      state.hasAcc = true;
    }
  },
  extraReducers:{
    [fetchPosts.pending]: (state, action)=>{
      state.status = 'loading'
    },
    [fetchPosts.fulfilled]:(state, action) =>{
      state.status = 'success'
      state.posts=action.payload;
    },
    [fetchPosts.rejected]: (state, action)=>{
      state.state = 'failed'
    }
  }
});

export const posts = (state) => state.home;
export const { toggleHasAcc } = homeSlice.actions;
export default homeSlice.reducer;
