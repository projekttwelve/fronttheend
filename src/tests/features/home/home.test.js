import React from 'react'
import { render, screen, cleanup } from '../../test-utils'
import home from '../../../features/home/home'

beforeEach(()=>{
  render(<home/>)
});

afterEach(()=>{
  cleanup()
})

test('Checking that sigIn component matches snapshot', () =>{
  const div = document.createElement('div')
  const component = render(<home/>, div)
  expect(component.container).toMatchSnapshot()
})
