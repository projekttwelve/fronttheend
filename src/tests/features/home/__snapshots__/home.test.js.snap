// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking that sigIn component matches snapshot 1`] = `
<div>
  <home />
</div>
`;
